apps/site/lib/site_web/controllers/schedule/finder_api.ex

```diff
@@ -450,14 +461,21 @@ defmodule SiteWeb.ScheduleController.FinderApi do
   defp maybe_remove_prediction_stop(nil), do: nil
   defp maybe_remove_prediction_stop(p), do: Map.put(p, :stop, nil)
 
-  @spec get_route_id(Route.id_t(), Trip.id_t()) :: Route.id_t()
+  @spec get_route_id(Route.id_t(), Trip.id_t()) :: Route.id_t() | nil
   defp get_route_id("Green", trip_id) do
-    schedule =
+    schedule_for_trip =
       trip_id
       |> Schedules.Repo.schedule_for_trip()
-      |> List.first()
 
-    schedule.route.id
+    if Enum.empty?(schedule_for_trip) do
+      nil
+    else
+      schedule =
+        schedule_for_trip
+        |> List.first()
+
+      schedule.route.id
```

Comment:
I think [CodeCov wants a new test to cover this piece](https://codecov.io/gh/mbta/dotcom/src/19b90a0c42932ac61a2fcb52d53156098e008cb6/apps/site/lib/site_web/controllers/schedule/finder_api.ex), maybe that'll make it pass?